Publish Premium Tiers blog post (#2782)
matrix-org/matrix.org: content/blog/2025/06/2025-06-11-announcing-premium-tiers.md

Today, only 50% of the costs of the Foundation are covered by its revenues (donations and memberships). 20% of the Foundation’s expenditure goes towards hosting the matrix.org free and public homeserver. If we add in the cost of the moderation work done by the Trust and Safety team, the total share of the costs attributable to the matrix.org homeserver account for 50% of all expenditure.

We’ve kept the matrix.org homeserver around,despite its costs, as we consider it essential to seed the network in support of the nurturing part of the Foundation’s mission: despite Matrix being decentralised by design, users need a trusted place to create a free Matrix account to try it out in the first place.

> The Matrix.org Foundation exists to act as a neutral custodian for Matrix and to nurture it as efficiently as possible as a single unfragmented standard, for the greater benefit of the whole ecosystem, not benefiting or privileging any single player or subset of players.

> However, we can’t continue to bear the cost of the server as is, and before we get to the extreme position of being forced to turn it off leaving its 370k monthly active users in the awkward position of finding a new home for their account, we’ve decided to try to alleviate some of these costs by setting-up a freemium offering and proposing premium plans in addition to the free ones. The goal is to get the server to an at least financially break-even position. If, by any chance, it was ending up profitable, the profit would directly be invested in [Trust and Safety](https://matrix.org/blog/2025/02/building-a-safer-matrix/), or other new programmes which can support the ecosystem. As a reminder, the Foundation is a Community Interest Company, i.e. a limited company which operates to provide a benefit to the community it serves rather than private profit.

Reviewer: ```suggestion
However, we can’t continue to bear the cost of the server as is, and before we get to the extreme position of being forced to turn it off leaving its 370k monthly active users in the awkward position of finding a new home for their account, we’ve decided to try to alleviate some of these costs by setting-up a freemium offering and proposing premium plans in addition to the free ones. The goal is to get the server to an at least financially break-even position. If, by any chance, it was ending up profitable, the profit would directly be invested in [Trust and Safety](/blog/2025/02/building-a-safer-matrix/), or other new programmes which can support the ecosystem. As a reminder, the Foundation is a Community Interest Company, i.e. a limited company which operates to provide a benefit to the community it serves rather than private profit.
```

We are limited in scope and design by the fact we need to ship a minimum viable product as soon as possible (we need to reduce costs now) and by not wanting to impose too much development (if any) to Matrix client developers.

Obviously we would have preferred to keep everything free of charge. We will never sell our users’data or cripple  our services with ads, so we need to find ethical sources of revenue. 

### When will the new plans take effect?

The roll-out will happen progressively, starting in the coming weeks and hopefully completing in the summer of 2025. We will start by opening up premium plans to new users only, before progressively migrating all existing accounts to a free plan which will give them the option to upgrade to a premium plan. Users will of course be notified ahead of  their account being migrated.

### Will this work with whatever Matrix client I use? 

> Yes. The plan management will be handled via the [My Account](https://account.matrix.org/account/) screens provided by the Matrix Authentication Service (MAS), and notifications to users will be sent in a dedicated room using the [Server Notices](https://spec.matrix.org/v1.14/client-server-api/#server-notices) feature built in to the Matrix protocol – already used by the homeserver to send automatic messages to the user – so should be seamless for every client. 

Reviewer: grammar nerd strikes again https://dictionary.cambridge.org/dictionary/english/build-in-into
```suggestion
Yes. The plan management will be handled via the [My Account](https://account.matrix.org/account/) screens provided by the Matrix Authentication Service (MAS), and notifications to users will be sent in a dedicated room using the [Server Notices](https://spec.matrix.org/v1.14/client-server-api/#server-notices) feature built into the Matrix protocol – already used by the homeserver to send automatic messages to the user – so should be seamless for every client. 
```

### I am a Matrix client developer, do I need to do anything?

> Assuming that your client already supports [Server Notices](https://spec.matrix.org/v1.14/client-server-api/#server-notices), the only complication introduced by the change is for applications distributed via the Apple App Store. Apple places [restrictions](https://developer.apple.com/app-store/review/guidelines/#in-app-purchase) on how payments are implemented by iOS (et al) apps.

Reviewer: overall that assumption will hold for most users, but in terms of spec, this is an optional feature and the spec does not require clients to implement it.

> Assuming that your client already supports [Server Notices]

Because of the above and since we also specifically point this out ("assuming"), the logical conclusion is to ask: what if not?

Reviewer 2: ```suggestion
There are two considerations from a Matrix client point of view:

- support for the [Server Notices](https://spec.matrix.org/v1.14/client-server-api/#server-notices) feature
- if the client is distributed via the Apple App Store, then support for [MSC4286](https://github.com/matrix-org/matrix-spec-proposals/pull/4286)

If the client doesn't show server notices at all then, whilst the client will remain usable with the matrix.org homeserver, your users will have a degraded UX as they won't receive notifications when encountering usage limits.

Apple places [restrictions](https://developer.apple.com/app-store/review/guidelines/#in-app-purchase) on how payments are implemented by iOS (et al) apps that are distributed via the App Store.
```

We expect that most, if not all, apps that fall within scope would be classified as what Apple calls “[Free Stand-alone Apps](https://developer.apple.com/app-store/review/guidelines/#free-stand-alone-apps)”. Such apps do not need to use in-app purchases so long as “there is no purchasing inside the app, or calls to action for purchase outside of the app”.

In order to meet these requirements we have proposed [MSC4286](https://github.com/matrix-org/matrix-spec-proposals/pull/4286) which provides a way for a homeserver (such as the matrix.org homeserver) to flag parts of messages as containing a call to action and for affected clients to be able to hide that content. Example implementations are linked in the MSC.

### I am already supporting the Foundation by paying an individual membership, will I have to pay for a premium plan too?

> No, individual members of the Foundation will get access to the premium features at no extra cost. This benefit will be implemented as part of the process of migrating existing accounts to the free plan. 

Reviewer: - could add link to membership page?
- should we already update the membership page to match? e.g. in this PR?

Reviewer: Assuming the redesign happens in time?
